addons/hr_expense/models/hr_expense.py

```diff
@@ -149,10 +159,30 @@ def _compute_state(self):
 
     @api.depends('quantity', 'unit_amount', 'tax_ids', 'currency_id')
     def _compute_amount(self):
-        for expense in self:
+        for expense in self.filtered(lambda e: e.product_has_cost or not e.product_id):
             taxes = expense.tax_ids.compute_all(expense.unit_amount, expense.currency_id, expense.quantity, expense.product_id, expense.employee_id.user_id.partner_id)
             expense.total_amount = taxes.get('total_included')
 
+    @api.depends('total_amount', 'tax_ids')
```

Comment:
I believe you should also add currency_id, employee_id and product_id to the depends

Reply:
I agree on currency_id, but I don't think  product_id and employee_id are needed :thinking: